### PR TITLE
feat: Global poller for updating worker status

### PR DESCRIPTION
**Pull request by shirish87**
Uses a shared poller to fetch and update the status of workers; helps reduce the no. of requests made to BrowserStack.

Ref #30

Let me know if there's scope for improvement or if an alternative approach is required.

### CODE REVIEW

**Comment by shirish87** [comment]
@Dignifiedquire, request you to review this PR. : )

File: worker.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      return (callback ? callback(err) : null)
+    }
+
+    updatedWorkers = updatedWorkers || []
+    var activeWorkerIds = updatedWorkers.map(function (worker) {
+      return worker.id
+    })
+
+    // process deletions
+    for (var i = 0, l = self.workers.length; i < l; i++) {
```
Comment (Dignifiedquire) [inline worker.js]: Please use `forEach` as well to stay consistent with the other code

File: worker.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      var worker = self.workers[i]
+      if (activeWorkerIds.indexOf(worker.id) === -1) {
+        self.unregisterWorker(worker)
+      }
+    }
+
+    // process updates
+    updatedWorkers.forEach(function (workerData) {
+      self.updateWorker(workerData)
```
Comment (Dignifiedquire) [inline worker.js]: Couldn't we merge this iteration with the one on l. 79?

File: worker.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+// expose a single, shared instance of WorkerManager
+var workerManager = new WorkerManager()
+
+module.exports = {
+  Worker: Worker,
+
+  WorkerManager: {
+    getInstance: function () {
+      return workerManager
+    }
```
Comment (Dignifiedquire) [inline worker.js]: Let's make this a bit simpler:

```js
module.exports = {
  Worker: Worker,
  getManager: function () {
    return workerManager
  }
}
```

File: worker.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    })
+  }
+}
+
+util.inherits(Worker, EventEmitter)
+
+/**
+ * Tracks worker state across runs.
+ */
+function WorkerManager () {
```
Comment (Dignifiedquire) [inline worker.js]: Let's put this into a separate file `worker-manager.js`

**Comment by Dignifiedquire** [comment]
Thanks for the ping, left some inline comments also this needs rebasing onto the latest `master`

**Comment by shirish87** [comment]
@Dignifiedquire, I've made some changes based on your suggestions. Please review. 
Thanks.

**Comment by Dignifiedquire** [comment]
Thanks, looking good :octocat:

**Pull request merged**